xmppserver/src/test/java/org/jivesoftware/openfire/session/LocalOutgoingServerSessionTest.java

```diff
@@ -0,0 +1,319 @@
+ */
+@ExtendWith(MockitoExtension.class)
+public class LocalOutgoingServerSessionTest
+{
+    private RemoteReceivingServerDummy remoteReceivingServerDummy;
+    private File tmpIdentityStoreFile;
+    private IdentityStore identityStore;
+    private File tmpTrustStoreFile;
+    private TrustStore trustStore;
+
+    /**
+     * Prepares the local server for operation. This mostly involves preparing the test fixture by mocking parts of the
+     * API that {@link LocalOutgoingServerSession#createOutgoingSession(DomainPair, int)} uses when establishing a
+     * connection.
+     */
+    @BeforeEach
+    public void setUpClass() throws Exception {
+        Fixtures.reconfigureOpenfireHome();
+        JiveGlobals.setProperty("xmpp.domain", Fixtures.XMPP_DOMAIN);
+        final XMPPServer xmppServer = Fixtures.mockXMPPServer();
+        XMPPServer.setInstance(xmppServer);
+
+        final File tmpDir = new File(System.getProperty("java.io.tmpdir"));
+
+        // Use a temporary file to hold the identity store that is used by the tests.
+        final CertificateStoreManager certificateStoreManager = mock(CertificateStoreManager.class, withSettings().lenient());
+        tmpIdentityStoreFile = new File(tmpDir, "unittest-identitystore-" + System.currentTimeMillis() + ".jks");
+        tmpIdentityStoreFile.deleteOnExit();
+        final CertificateStoreConfiguration identityStoreConfig = new CertificateStoreConfiguration("jks", tmpIdentityStoreFile, "secret".toCharArray(), tmpDir);
+        identityStore = new IdentityStore(identityStoreConfig, true);
+        doReturn(identityStore).when(certificateStoreManager).getIdentityStore(any());
+
+        // Use a temporary file to hold the trust store that is used by the tests.
+        tmpTrustStoreFile = new File(tmpDir, "unittest-truststore-" + System.currentTimeMillis() + ".jks");
+        tmpTrustStoreFile.deleteOnExit();
+        final CertificateStoreConfiguration trustStoreConfig = new CertificateStoreConfiguration("jks", tmpTrustStoreFile, "secret".toCharArray(), tmpDir);
+        trustStore = new TrustStore(trustStoreConfig, true);
+        doReturn(trustStore).when(certificateStoreManager).getTrustStore(any());
+
+        // Mock the connection configuration.
+        doReturn(certificateStoreManager).when(xmppServer).getCertificateStoreManager();
+
+        final ConnectionManager connectionManager = Fixtures.mockConnectionManager();
+        final ConnectionListener connectionListener = Fixtures.mockConnectionListener();
+        doAnswer(new ConnectionConfigurationAnswer(identityStoreConfig, trustStoreConfig)).when(connectionListener).generateConnectionConfiguration();
+        doReturn(Set.of(connectionListener)).when(connectionManager).getListeners(any(ConnectionType.class));
+        doReturn(connectionListener).when(connectionManager).getListener(any(ConnectionType.class), anyBoolean());
+        doReturn(connectionManager).when(xmppServer).getConnectionManager();
```

Comment:
As above - do we need these explicit mock behaviours?

Reply:
Yes, they're installing test-specific trust material. Tests fail badly without them.